# Subject: [PATCH] [OTNS] remove -Werror build flag in [mbedtls] to enable
# building legacy code on newer (Apple Clang) compilers.
# Also sets the minimum CMake version for mbedtls from 2.8.12 to 3.5.1, so
# that a newer CMake (4.x) can still build the library.
# This patch was made for commit 0a5152b4fa9f9cbff57da89d9aa33d409e915241 (v11)


diff --git a/third_party/mbedtls/repo/CMakeLists.txt b/third_party/mbedtls/repo/CMakeLists.txt
index ac24bf41b..508bbfba7 100644
--- a/third_party/mbedtls/repo/CMakeLists.txt
+++ b/third_party/mbedtls/repo/CMakeLists.txt
@@ -20,7 +20,7 @@
 #   mbedtls, mbedx509, mbedcrypto and apidoc targets.
 #

-cmake_minimum_required(VERSION 2.8.12)
+cmake_minimum_required(VERSION 3.5.1)
 if(TEST_CPP)
     project("mbed TLS" C CXX)
 else()
@@ -218,7 +218,7 @@ if(MBEDTLS_FATAL_WARNINGS)
     endif(CMAKE_COMPILER_IS_MSVC)

     if(CMAKE_COMPILER_IS_CLANG OR CMAKE_COMPILER_IS_GNU)
-        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+        # set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
         if(UNSAFE_BUILD)
             set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Wno-error=cpp")
             set(CMAKE_C_FLAGS_ASAN "${CMAKE_C_FLAGS_ASAN} -Wno-error=cpp")
